src/index.js

```diff
@@ -29,18 +29,22 @@ export default class ReactPasswordStrength extends React.Component {
     });
   }
 
-  checkValidity(score, length) {
-    const { minScore, minLength } = this.props;
-    return score >= minScore && length >= minLength;
-  }
-
   handleChange(e) {
-    const { changeCallback } = this.props;
+    const { changeCallback, minLength, minScore } = this.props;
     const password = this.refs['ReactPasswordStrength-input'].value;
-    const { score } = zxcvbn(password);
+
+    var score;
+
+    // always sets a zero score when min length requirement is not met
+    // which avoids unnecessary zxcvbn computations (they require quite lots of CPU)
+    if (minLength && password.length < minLength) {
+      score = 0
```

Comment:
What I mean is we never need to check that `minLength` is defined because it will always be defined. This is because it has a default value (so it will never be undefined) and it has type-checking (so it will always be a number).